src/plugins/crypto/gpg.c

```diff
@@ -327,18 +381,10 @@ int elektraCryptoGpgCall (KeySet * conf, Key * errorKey, Key * msgKey, char * ar
 	argv[argc - 1] = NULL;
 
 	// check that the gpg binary exists and that it is executable
-	if (access (argv[0], F_OK))
+	if (isExecutable (argv[0], errorKey) < 0)
 	{
-		ELEKTRA_SET_ERRORF (ELEKTRA_ERROR_CRYPTO_GPG, errorKey, "gpg binary %s not found", argv[0]);
 		elektraFree (argv[0]);
-		return -1;
-	}
-
-	if (access (argv[0], X_OK))
-	{
-		ELEKTRA_SET_ERRORF (ELEKTRA_ERROR_CRYPTO_GPG, errorKey, "gpg binary %s has no permission to execute", argv[0]);
-		elektraFree (argv[0]);
```

Comment:
It is freed or am I mistaking?

	// check that the gpg binary exists and that it is executable
	if (isExecutable (argv[0], errorKey) < 0)
	{
		elektraFree (argv[0]);
		return -1; // error set by isExecutable()
	}